File: server/BooksStorage/DTOs/Book/BookCreateDTO.cs
using System.ComponentModel.DataAnnotations;

namespace BooksStorage.DTOs.Book;

public class BookCreateDto
{
    [Required] public string Name { get; set; }
    [Required] public string Author { get; set; }

    [Required]
    [Display(Name = "Year of Publication")]
    public int PublicationYear { get; set; }

    [Required] public decimal Price { get; set; }
    [Required] public string Category { get; set; }
}

File: client/src/app/models/book-create-dto.ts
export class BookCreateDto {
    public name: string;
    public author: string;
    public publicationYear: number;
    public price: number;
    public category: string;
    public email: string;

    constructor(
        name: string = '',
        author: string = '',
        yearOfPublication: number = 0o000,
        price: number = 0,
        category: string = '',
        email: string = ''
      ) {
        this.name = name;
        this.author = author;
        this.publicationYear = yearOfPublication;
        this.price = price;
        this.category = category;
        this.email = email;
      }
}
